haskell/cabal.bzl

```diff
@@ -116,6 +117,7 @@ def _prepare_cabal_inputs(hs, cc, posix, dep_info, cc_info, component, package_i
     ]
     args.add_all([component, package_id, setup, cabal.dirname, package_database.dirname])
     args.add("--flags=" + " ".join(flags))
+    args.add_all(sorted(compiler_flags), format_each = "--ghc-option=%s")
```

Comment:
Why sort the flags? Is that generally safe?

Reply:
I was trying to avoid recompilation on cosmetic reordering of flags, but you are right, this could change the meaning of the flags! Will fix